server/topic.go

```diff
@@ -1026,20 +1036,20 @@ func (t *Topic) handleBroadcast(msg *ServerComMessage) {
 		// Send message to session.
 		if !sess.queueOut(msg) {
 			logs.Warn.Printf("topic[%s]: connection stuck, detaching - %s", t.name, sess.sid)
-			// The whole session is being dropped, so sessionLeave.pkt is not set.
-			// Must not block here: it may lead to a deadlock.
-			select {
-			case t.unreg <- &sessionLeave{sess: sess}:
-			default:
-				logs.Err.Printf("topic[%s]: unreg queue full - %s", t.name, sess.sid)
-			}
+			dropSessions = append(dropSessions, sess)
 		}
 	}
 
 	if !t.isProxy && pushRcpt != nil {
 		// usersPush will update unread message count and send push notification.
 		usersPush(pushRcpt)
 	}
+
+	// Drop "bad" sessions.
+	for _, sess := range dropSessions {
+		// The whole session is being dropped, so sessionLeave.pkt is not set.
+		t.processLeaveRequest(&sessionLeave{sess: sess})
```

Comment:
Is this correct? `t.processLeaveRequest` calls `t.handleLeaveRequest` and that calls `t.processLeaveRequest` again.

Reply:
It would also be nice to make names `processLeaveRequest` and `handleLeaveRequest` more different. And add more descriptive comments. Hard to tell what each one of them does differently from the other.

Reply:
Renamed `unregisterSession`.